### ISSUE TITLE
Uninitialized variables don't raise an exception on access 

### ISSUE THREAD

**Issue by apblack** [gracelang/minigrace#113]
Whereas making a request on an uninitialized variable correctly raises a ProgrammingError:

```
ProgrammingError at line 2 of main: Requested method '+' on uninitialised value
```

accessing an uninitialized variable in other ways, such as using it as an argument, does not raise the exception.  tests/known-failing/uninitialized_test.grace  demonstrates the issue.

There should be an exception

```
def UninitializedVariable = ProgrammingError.refine "UninitializedVariable"
```

that is used for these errors.

**Comment by apblack** [comment]
Accessor methods compiled for an object should check for uninitialized variables before returning — but _only_ if the variable concerned is not initialized in its declaration.   Once the variable is initialized, perhaps the accessor method could be overwritten with a new one that no longer does the check.  This could be accomplished by a setter that deletes itself.

**Comment by apblack** [comment]
This issue was fixed in commit be23aa1.
